lib/railway_ipc/consumer/consumer.rb

```diff
@@ -0,0 +1,58 @@
+require "json"
+require "base64"
+require "railway_ipc/consumer/consumer_response_handlers"
+
+module RailwayIpc
+  class Consumer
+    include Sneakers::Worker
+    attr_reader :message, :handler
+
+    def self.listen_to(queue:, exchange:)
+      from_queue queue,
+                 exchange: exchange,
+                 durable: true,
+                 exchange_type: :fanout,
+                 connection: RailwayIpc.bunny_connection
+    end
+
+    def self.handle(message_type, with:)
+      ConsumerResponseHandlers.instance.register(message: message_type, handler: with)
```

Comment:
`Handlers` classes are used to store global state for Consumers and RPC participants. The way the gem is setup, there are 3 total `Handlers` per running Ruby process (Client, Server, and Consumer). This _could_ be a limitation of this implementation if we have a ruby process with several Consumers or Servers running at once. Will have to confirm with @SophieDeBenedetto on how likely this is.